### DISCUSSION TITLE
Rip doesn't return a number line in vim system()?

### DISCUSSION THREAD

**Discussion by rickswe** [BurntSushi/ripgrep#1780] [General] [answered]
Hello, when in vim I can use grep and get the expected result but I can't use rip grep and get the same result. When I do it on the shell rip grep works fine and returns 8882 but not in vim. I'm trying to get a line number where the pattern was found and put it in a vim variable or register. 

This works
```vim
let @*=system("grep -n '^setpos' '/usr/local/share/vim/vim82/doc/tags' ")
"Output --> 8882:setpos()	eval.txt	/*setpos()*
```
This does not work. No number.
```vim
let @*=system("rg '^setpos' '/usr/local/share/vim/vim82/doc/tags'")
"Output --> setpos()	eval.txt	/*setpos()*
```
For now I'm just using grep on this part of the code. Thanks

**Comment by BurntSushi** [comment, marked as answer]
Use `-n`, just like grep.

When ripgrep is run with a tty, it automatically enables certain options like colors, line numbers and file headings. But when without a tty (like inside vim), it runs more like grep. So just pass the flags you need.

**Reply by rickswe** [reply]
Thanks, that worked ! Appreciate your help. :)
